README.md

```diff
@@ -15,8 +15,43 @@ Run the following commands:
 go get github.com/moesif/moesifmiddleware-go
 ```
 
-Edit moesif_options/moesif_options.go to change the application id to your application id obtained from Moesif.
+## Configuration options
 
+Please note the request is the original http.Request and the response is moesifResponseRecorder.
+
+```go
+// Moesif Response Recorder
+ type MoesifResponseRecorder struct {
+	 rw http.ResponseWriter
+	 status int
+	 writer io.Writer
+	 header map[string][]string
+ }
+```
+
+#### __`Application_Id`__
+(__required__), _string_, is obtained via your Moesif Account, this is required.
+
+#### __`Should_Skip`__
+(optional) _(request, response) => boolean_, a function that takes a request and a response,
+and returns true if you want to skip this particular event.
+
+#### __`Identify_User`__
+(optional, but highly recommended) _(request, response) => string_, a function that takes a request and response, and returns a string that is the user id used by your system. While Moesif tries to identify users automatically, but different frameworks and your implementation might be very different, it would be helpful and much more accurate to provide this function.
+
+#### __`Get_Metadata`__
+(optional) _(request, response) => dictionary_, a function that takes a request and response, and
+returns a dictionary (must be able to be encoded into JSON). This allows your
+to associate this event with custom metadata. For example, you may want to save a VM instance_id, a trace_id, or a tenant_id with the request.
+
+#### __`Get_Session_Token`__
+(optional) _(request, response) => string_, a function that takes a request and response, and returns a string that is the session token for this event. Again, Moesif tries to get the session token automatically, but if you setup is very different from standard, this function will be very help for tying events together, and help you replay the events.
```

Comment:
a function that takes a request and response, and returns a string that is the session token for this event. Moesif tries to get the session token automatically, but if this doesn't work for your service, you should use this to identify sessions.